### PR TITLE
Trigger

### PR DESCRIPTION
**Pull request by yican95**

### CODE REVIEW

File: src/main/resources/data.sql [hunk 1]
```diff
@@ -15,7 +15,13 @@ DROP TABLE IF EXISTS Orders CASCADE;
 DROP TABLE IF EXISTS Reviews CASCADE;
 
 DROP TRIGGER IF EXISTS addSpecificUserTrigger ON Users;
+DROP TRIGGER IF EXISTS hashPasswordTrigger ON Users;
 DROP TRIGGER IF EXISTS updateCustomerRewardPointsTrigger ON Orders;
+DROP TRIGGER IF EXISTS checkAcceptedOrdersTrigger ON Orders CASCADE;
+DROP TRIGGER IF EXISTS checkPartTimeRiderShiftTrigger ON PTShifts CASCADE;
+
+CREATE EXTENSION IF NOT EXISTS pgcrypto;
```
Comment (GabrielYik) [inline src/main/resources/data.sql:23]: Do we need a `DROP EXTENSION IF EXISTS` for pgcrypto?

Comment (GabrielYik) [inline src/main/resources/data.sql:23]: Do we need a `DROP EXTENSION IF EXISTS` for pgcrypto?

File: src/main/resources/data.sql [hunk 2]
```diff
@@ -165,6 +171,40 @@ CREATE TRIGGER addSpecificUserTrigger
...
+			WHERE O.drid = NEW.drid AND O.deliveryTime IS NULL AND O.oid <> NEW.oid;
+        IF aoid IS NOT NULL THEN 
+			RAISE exception '% has already accepted %', NEW.drid, aoid;
+		END IF;
+		RETURN NULL;
+    END;
+$$ LANGUAGE PLPGSQL;
+
+CREATE CONSTRAINT TRIGGER checkAcceptedOrdersTrigger
+    AFTER UPDATE OF drid OR INSERT
```
Comment (GabrielYik) [inline src/main/resources/data.sql:203]: Not before?

File: src/main/resources/data.sql [hunk 3]
```diff
@@ -180,6 +220,28 @@ CREATE TRIGGER updateCustomerRewardPointsTrigger
...
+			((NEW.startHour - P.endHour >= 1) OR (P.startHour - NEW.endHour >= 1));
+        IF shift IS NOT NULL THEN 
+			RAISE exception 'Need atleast 1 hour break interval';
+		END IF;
+		RETURN NULL;
+    END;
+$$ LANGUAGE PLPGSQL;
+
+CREATE CONSTRAINT TRIGGER checkPartTimeRiderShiftTrigger
+    AFTER UPDATE OR INSERT
```
Comment (GabrielYik) [inline src/main/resources/data.sql:240]: Not before?

File: src/main/java/foodscooter/repositories/JdbcUsersRepository.java [hunk 4]
```diff
@@ -48,7 +48,7 @@ public int add(String username, String password, UserType userType) {
     return jdbcTemplate.queryForObject(
       "SELECT * "
       + "FROM Users U "
-      + "WHERE U.username = ? AND U.password = ? ;",
+      + "WHERE U.username = ? AND U.password = digest(?, 'sha1') ;",
```
[the remark this answers is not in the document]
Comment (yican95) [reply src/main/java/foodscooter/repositories/JdbcUsersRepository.java:51]: this one only the insert or update will hash. we comparing the user input

**Pull request merged**